### DISCUSSION TITLE
프론트엔드 기초 아이디어

### DISCUSSION THREAD

**Discussion by Circlewee** [heading1/WYLSBingsu#1] [Ideas]
# 구현
1. 문법
   - [x] Vanila JS  
   - [x] JSX  
2. 컴파일러
   - [x] Babel (JSX를 사용한다면 필요)
3. 번들러
   - [x] Webpack
   - [ ] parcel
   - [ ] ESBuild
 4. 스타일  
    - [x] basic css + webpack 설정으로 각각의 css파일을 컴파일링해 클래스의 중복을 막는다. [Reference](https://velog.io/@jaeung5169/webpack-%EC%82%AC%EC%9A%A9%EC%8B%9C-class-name-%EC%A4%91%EB%B3%B5-%ED%94%BC%ED%95%98%EA%B8%B0)
 5. 구현방식
    - [x] 함수형 컴포넌트

# 페이지 리스트
1. 로그인
2. 초기 로그인 후 닉네임 설정 화면
3. 공유 링크 알려주는 화면
4. 사용자가 편지 작성하는 화면
5. 사용자가 편지 확인하는 화면
6. 마이페이지 (내 링크 목록들?)

**Comment by jinux127** [comment]
웹 컴포넌트 방식의 개발을 어떻게 진행할까요?

## 참고 사이트
1. [Javascript 로 React 따라잡기](https://junilhwang.github.io/TIL/Javascript/Design/Vanilla-JS-Component/#_1-%E1%84%8F%E1%85%A5%E1%86%B7%E1%84%91%E1%85%A9%E1%84%82%E1%85%A5%E1%86%AB%E1%84%90%E1%85%B3%E1%84%8B%E1%85%AA-%E1%84%89%E1%85%A1%E1%86%BC%E1%84%90%E1%85%A2%E1%84%80%E1%85%AA%E1%86%AB%E1%84%85%E1%85%B5)
2. [나만의 리액트 라이브러리 만들기](https://bluewings.github.io/build-your-own-react/)
3. [바닐라 JS에서 SPA 라우팅 시스템 구현하기](https://kdydesign.github.io/2020/10/06/spa-route-tutorial/)
